### PR TITLE
fix(viewer): viewer crd controller panics using tensorboard image without tag. Part of #5471

### PR DESCRIPTION
**Pull request by Bobgy**
**Description of your changes:**
Part of #5471 

When testing https://github.com/kubeflow/pipelines/issues/5471, I found viewer crd controller crashes with tensorboard image without tag. We should treat them as tensorflow v2.

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: backend/src/crd/controller/viewer/reconciler/reconciler.go [hunk 1]
```diff
@@ -178,15 +178,23 @@ func setPodSpecForTensorboard(view *viewerV1beta1.Viewer, s *corev1.PodSpec) {
...
 		fmt.Sprintf("--path_prefix=/tensorboard/%s/", view.Name),
 	}
-
-	tfImageVersion := strings.Split(view.Spec.TensorboardSpec.TensorflowImage, ":")[1]
-	// This is needed for tf 2.0
-	if !strings.HasPrefix(tfImageVersion, `1.`) {
+	isTensorflowV1 := false
+	parts := strings.Split(view.Spec.TensorboardSpec.TensorflowImage, ":")
+	// an image might not contain a tag
+	if len(parts) == 2 {
```
Comment (capri-xiyue) [inline backend/src/crd/controller/viewer/reconciler/reconciler.go:184]: What is the case if len(parts) !=2, is that invalid? Should we return error if len(parts) !=2 ?

Comment (Bobgy) [reply backend/src/crd/controller/viewer/reconciler/reconciler.go:184]: e.g. an image without tag is used, it's valid and I think we can now assume it's tensorflow 2+

**Pull request merged**